kafka-client-producer-example/src/main/java/com/example/producer/client/kafka/producerforstreams/IncrementalKeyProducer.java

```diff
@@ -0,0 +1,42 @@
+package com.example.producer.client.kafka.producerforstreams;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.serialization.Serdes;
+
+import java.util.Properties;
+import java.util.concurrent.Future;
+import java.util.stream.IntStream;
+
+public class IncrementalKeyProducer {
+    public static void main(String[] args) {
+        // configuration
+        Properties properties = new Properties();
+        properties.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "127.0.0.1:9092");
+        properties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, Serdes.Integer().serializer().getClass());
+        properties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, Serdes.String().serializer().getClass());
+
+        KafkaProducer<Integer, String> producer = new KafkaProducer<>(properties);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *producer*. Currently, there are execution paths that do not contain closure statements, for example, when an else condition at line 24 evaluates to true. Ensure *producer* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).